libs/hdf-converters/src/veracode-mapper.ts

```diff
@@ -0,0 +1,519 @@
+      ) as Record<string, unknown>[]) {
+        vulnarr.push(vuln);
+      }
+    } else {
+      vulnarr.push(
+        _.get(component, `vulnerabilities.vulnerability`) as Record<
+          string,
+          unknown
+        >
+      );
+    }
+  }
+  return vulnarr;
+}
+
+function componentlistcreate(input:unknown){
+  const componentlist: Record<string, unknown>[] =[]
+  if (isArray(_.get(input, `component`))) {
+    for (const value of _.get(input, `component`) as Record<
+      string,
+      unknown
+    >[]) {
+      if (_.get(value, `vulnerabilities`) !== '') {
+        componentlist.push(value);
+      }
+    }
+  } else {
+    componentlist.push(_.get(input, 'component') as Record<string, unknown>);
+  }
+  return componentlist
+}
+
+function componentTransform(input: unknown) {
+  const componentlist: Record<string, unknown>[] = componentlistcreate(input);
+
+  const vulnarr: Record<string, unknown>[] = vulnarrcreate(componentlist);
+
+  vulnarr.forEach((vuln) => {
+    const components = [];
+    let location = '';
+    const currcve = _.get(vuln, `cve_id`);
+    const impact = impactMapping(_.get(vuln, `severity`) as number);
+    _.set(vuln, `impact`, impact);
+    for (const component of componentlist) {
+      let hascve = false;
+      if (Array.isArray(_.get(component, `vulnerabilities.vulnerability`))) {
+        (_.get(
+          component,
+          `vulnerabilities.vulnerability`
+        ) as Record<string, unknown>[]).forEach((compcve) => {
+          if (_.get(compcve, 'cve_id') === currcve) {
+            hascve = true;
+            location += ` ${_.get(component, FILE_PATH_VALUE)}`;
+            _.set(vuln, `paths`, location);
+          }
+        })
+      } else {
+        if (
+          _.get(component, 'vulnerabilities.vulnerability.cve_id') === currcve
+        ) {
+          hascve = true;
+          location += ` ${_.get(component, FILE_PATH_VALUE)}`;
+          _.set(vuln, `paths`, location);
+        }
+      }
+      if (hascve === true) {
+        const proxy = _.cloneDeep(component);
+        components.push(_.omit(proxy, `vulnerabilities`));
+      }
+    }
+    _.set(vuln, `components`, components);
+  })
+  return vulnarr as unknown;
+}
+
+function controlMappingCve(): MappedTransform<
+  ExecJSON.Control & ILookupPath,
+  ILookupPath
+> {
+  return {
+    id: {path: 'cve_id'},
+    title: {path: 'cve_id'},
+    desc: {path: 'cve_summary'},
+    impact: {path: 'impact'},
+    refs: [],
+    tags: {
+      cwe: {path: 'cwe_id'},
+      nist: {
+        path: 'cwe_id',
+        transformer: (value: string) => {
+          if (isArray(value)) {
+            return CWE_NIST_MAPPING.nistFilter(
+              value as string[],
+              DEFAULT_NIST_TAG
+            );
+          } else {
+            return CWE_NIST_MAPPING.nistFilter([value], DEFAULT_NIST_TAG);
+          }
+        }
+      }
+    },
+    source_location: {
+      ref: {path: 'paths'}
+    },
+
```

Comment:
also make sure to run the linter (yarn run lint) which should autofix most of the issues and complain about the places where it can't fix them.  i would run it as a last step before pushing up to the repo so please don't use it as an easy first win type of commit cause you're probably gonna have to do it again at the end anyways lol